core/include/gnuradio-4.0/Block.hpp

```diff
@@ -876,10 +875,30 @@ struct Block : protected std::tuple<Arguments...> {
         if (numerator != 1UZ || denominator != 1UZ) {
             // TODO: this ill-defined checks can be done only once after parameters were changed
             const double ratio          = static_cast<double>(numerator) / static_cast<double>(denominator);
-            bool         is_ill_defined = (denominator > ports_status.in_max_samples) || (static_cast<double>(ports_status.in_min_samples) * ratio > static_cast<double>(ports_status.out_max_samples))
+            bool         is_ill_defined = (denominator > ports_status.in_max_samples)                                                            //
+                               || (static_cast<double>(ports_status.in_min_samples) * ratio > static_cast<double>(ports_status.out_max_samples)) //
                                || (static_cast<double>(ports_status.in_max_samples) * ratio < static_cast<double>(ports_status.out_min_samples));
-            assert(!is_ill_defined);
+
+            if (denominator > ports_status.in_max_samples) { // TODO: convert to proper error message send to msgOut
+                fmt::println(stderr, "configuration error for block {}: denominator {} > ports_status.in_max_samples {}", name, denominator, ports_status.in_max_samples);
+                assert(false && "denominator needs to be <= max InPort sample constraints");
+                return work::Status::ERROR;
+            }
+            if (static_cast<double>(ports_status.in_min_samples) * ratio > static_cast<double>(ports_status.out_max_samples)) { // TODO: convert to proper error message send to msgOut
+                fmt::println(stderr, "configuration error for block {}:  ports_status.in_min_samples * ratio {} > ports_status.out_max_samples {}, ratio (num {}/den {} = {})", name,
+                             static_cast<double>(ports_status.in_min_samples) * ratio, static_cast<double>(ports_status.out_max_samples), numerator, denominator, ratio);
+                assert(false && "reduced min-required input sample needs to be <= max OutPort sample constraints");
+                return work::Status::ERROR;
+            }
+            if (static_cast<double>(ports_status.in_max_samples) * ratio < static_cast<double>(ports_status.out_min_samples)) { // TODO: convert to proper error message send to msgOut
+                fmt::println(stderr, "configuration error for block {}:  ports_status.in_max_samples * ratio {} > ports_status.out_min_samples {}, ratio (num {}/den {} = {})", name,
+                             static_cast<double>(ports_status.in_max_samples) * ratio, static_cast<double>(ports_status.out_min_samples), numerator, denominator, ratio);
+                assert(false && "reduced max-required input sample needs to be <= min OutPort sample constraints");
+                return work::Status::ERROR;
+            }
```

Comment:
Does it makes sense to move all settings/parameter checking to some function like `checkSettings` and it should be called only if parameters are changed inside `settingsChanged`? Some additional ill-defined parameters checking can be also then added.

Reply:
Yes, it should be part of the next refactoring PR. I would leave it as is until we settled on the other items.